lib/src/Uncertainty/Process/GaussianProcess.cxx

```diff
@@ -227,6 +227,16 @@ void GaussianProcess::setTimeGrid(const RegularGrid & timeGrid)
   setMesh(timeGrid);
 }
 
+GaussianProcess::SamplingMethod GaussianProcess::getSamplingMethod() const
+{
+  if (samplingMethod_ == 1)
+    return SamplingMethod::HMAT;
+  else if (samplingMethod_ == 2)
+    return SamplingMethod::GIBBS;
+  else
+    return SamplingMethod::CHOLESKY;
+}
+
```

Comment:
cant we just cast samplingMethod_ into the enum type ?

Reply:
you mean something like 
```cxx
GaussianProcess::SamplingMethod GaussianProcess::getSamplingMethod() const
{
  return samplingMethod_;
}

```

Reply:
maybe the compiler wants a static_cast ?

Reply:
indeed